tools/diff-processor/detector/detector.go

```diff
@@ -152,3 +166,67 @@ func suggestedTest(resourceName string, untested []string) string {
 	}
 	return strings.ReplaceAll(string(f.Bytes()), `"VALUE"`, "# value needed")
 }
+
+func DetectMissingDocs(schemaDiff diff.SchemaDiff, repoPath string) (map[string]MissingDocDetails, error) {
+	ret := make(map[string]MissingDocDetails)
+	for resource, resourceDiff := range schemaDiff {
+		docFilePath := resourceToDocFile(resource, repoPath)
+		var argumentsInDoc, attributesInDoc map[string]bool
+
+		details := MissingDocDetails{
+			FilePath: strings.ReplaceAll(docFilePath, repoPath, ""),
+		}
+		content, err := os.ReadFile(docFilePath)
+		if err != nil && !os.IsNotExist(err) {
+			return nil, fmt.Errorf("failed to read resource doc %s: %w", docFilePath, err)
+		}
+		if err == nil {
+			parser := documentparser.NewParser()
+			err = parser.Parse(content)
+			if err != nil {
+				return nil, fmt.Errorf("failed to pass document %s: %w", docFilePath, err)
+			}
+			argumentsInDoc = listToMap(parser.Arguments())
+			attributesInDoc = listToMap(parser.Attributes())
+		}
+
+		for field, fieldDiff := range resourceDiff.Fields {
+			if !isNewField(fieldDiff) {
+				continue
+			}
+			if isAttribute(fieldDiff) {
+				if !attributesInDoc[field] {
+					details.Fields = append(details.Fields, MissingDocField{Field: field, Section: "Attributes Reference"})
+				}
+			} else if !argumentsInDoc[field] {
+				details.Fields = append(details.Fields, MissingDocField{Field: field, Section: "Arguments Reference"})
+			}
+		}
+		if len(details.Fields) > 0 {
+			ret[resource] = details
+		}
+	}
+	return ret, nil
+}
+
+func isAttribute(fieldDiff diff.FieldDiff) bool {
```

Comment:
I might be reading this wrong - it seems that we still count the non-top level computed fields as attributes?

I’m starting to wonder that even if we only run the missing doc detector on newly added fields, inconsistencies in how attributes and arguments are divided in existing resource docs could still lead to issues(?), like for subfields added to existing objects. What are your thoughts on this?

Reply:
Decided to not differentiate between argument and attribute, since finding some fields not conforming to that rule, eg. terraform_labels.